pkg/k8shandler/collection.go

```diff
@@ -366,3 +368,23 @@ func isBufferFlushRequired(current *apps.DaemonSet, desired *apps.DaemonSet) boo
 func getServiceAccountLogCollectorUID() types.UID {
 	return serviceAccountLogCollectorUID
 }
+
+func envValueEqual(cenv, denv []core.EnvVar) bool {
```

Comment:
There should be a unit test for this.  Wouldn't we be better to implement a whitelist of env vars that we simply skip checking?  Maybe rebuild the actual list as needed.

Reply:
I've added a unit test.   Checking the EnvVar array as is, though...  Do you think it's ok?  Or better to rebuild a list?